cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/Executor.java

```diff
@@ -242,27 +280,33 @@ private void moveReplicas() {
       }
       // After the partition movement finishes, wait for the controller to clean the reassignment zkPath. This also
       // ensures a clean stop when the execution is stopped in the middle.
-      while (!_executionTaskManager.inExecutionTasks().isEmpty()) {
-        LOG.info("Waiting for {} tasks to finish: {}", _executionTaskManager.inExecutionTasks().size(),
-                 _executionTaskManager.inExecutionTasks());
+      Set<ExecutionTask> inExecutionTasks = _executionTaskManager.inExecutionTasks();
+      while (!inExecutionTasks.isEmpty()) {
+        LOG.info("Waiting for {} tasks moving {} MB to finish: {}",
+                 inExecutionTasks.size(), inExecutionDataToMoveInMB(inExecutionTasks), inExecutionTasks);
         waitForExecutionTaskToFinish();
+        inExecutionTasks = _executionTaskManager.inExecutionTasks();
       }
       if (_executionTaskManager.inProgressTasks().isEmpty()) {
         LOG.info("Partition movements finished.");
-      } else if (_stopRequested) {
-        LOG.info("Partition movements stopped. {} in progress, {} pending, {} aborted, {} dead.",
-                 _executionTaskManager.inProgressTasks().size(),
-                 _executionTaskManager.remainingPartitionMovements().size(),
-                 _executionTaskManager.abortedTasks().size(),
-                 _executionTaskManager.deadTasks().size());
+      } else if (_stopRequested.get()) {
+        ExecutionTaskManager.ExecutionState executionState = _executionTaskManager.getExecutionTasksSummary();
+        LOG.info("Partition movements stopped. {} in-progress, {} pending, {} aborting, {} aborted, {} dead, "
+                 + "{} remaining data to move.",
+                 executionState.inProgressTasks().size(),
+                 executionState.remainingPartitionMovements().size(),
+                 executionState.abortingTasks(),
+                 executionState.abortedTasks().size(),
+                 executionState.deadTasks().size(),
+                 executionState.remainingDataToMoveInMB());
       }
     }
 
-    private void moveLeaders() {
+    private void moveLeaderships() {
```

Comment:
Orthogonal to this patch and not urgent. We may want to expose the leadership movement to the users as well. This could be a separate patch, though.

Reply:
Thanks for the feedback!
Is this ticket relevant to your request -> https://github.com/linkedin/cruise-control/issues/159, or is this some other functionality?

Reply:
It is something different. I was thinking about the leadership movement progress.

Reply:
Yes, that's definitely something we should add. I will create an issue about it for tracking purposes.